### PR TITLE
Added unit tests to code related with annotated resources

### PR DESCRIPTION
**Pull request by fgiorgetti**
As I am working on a new integration test (blackbox) that
will validate skupper handling annotated resources, I ended
up creating unit tests for most of the functions used by
the related code.

### CODE REVIEW

File: cmd/service-controller/definition_monitor_test.go [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+	type result struct {
+		service types.ServiceInterface
+		success bool
+	}
+
+	type test struct {
+		name     string
+		service  *corev1.Service
+		expected result
+	}
```
Comment (nicob87) [inline cmd/service-controller/definition_monitor_test.go:196]: Another minor comment:
In all the test cases it seems to be a
```type test struct {
		name     string
		service  *corev1.Service
		expected result
	}
```
or a:
```
type test struct {
		name       string
		deployment *v1.Deployment
		expected   result
	}
```

maybe, you can just create two globals:
testDeployment and testService

File: pkg/kube/configmaps_test.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+			Namespace: NS,
+			OwnerReferences: []metav1.OwnerReference{
+				{Name: "TestNewConfigMap"},
+			},
+		},
+	}
+	kubeClient.CoreV1().ConfigMaps(NS).Create(existingCm)
+
+	// Add a fake reaction when trying to get or create "error-cm"
+	kubeClient.Fake.PrependReactor("*", "configmaps", func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
```
Comment (nicob87) [inline pkg/kube/configmaps_test.go:50]: Nice!

**Comment by nicob87** [comment]
I understand CI had not run for this PR because of the "ci not running for fork with ci enabled in the fork also", would be nice to trigger ci before merging.
Unfortunately, the only way of doing that is "pushing a commit" (after unfollowing skupper project in the fork), you can reset the commit and force a push, as a way of triggering the ci, in the same pr, without adding additional junk commits.

File: cmd/service-controller/definition_monitor_test.go (hunk 1 above)
Comment (fgiorgetti) [reply cmd/service-controller/definition_monitor_test.go:196]: My idea was to keep these inner types contained and private to the
test method, as it is mostly suited to those particular unit tests.

**Comment by fgiorgetti** [comment]
> I understand CI had not run for this PR because of the "ci not running for fork with ci enabled in the fork also", would be nice to trigger ci before merging.
> Unfortunately, the only way of doing that is "pushing a commit" (after unfollowing skupper project in the fork), you can reset the commit and force a push, as a way of triggering the ci, in the same pr, without adding additional junk commits.

Correct. I have some small changes to push and now that I have unfollowed skupper on my fork, I suppose it is going to run.

**Comment by fgiorgetti** [comment]
> Nice test additions.
> 
> Suggest updating Makefile so that these can be run in ci, or we can do it in a separate pr
> 
> pkg-test:
> go test -v -count=1 ./pkg
> 
> service-controller-mock-test:
> go test -v count=1 ./cmd/service-controller
> 
> service-controller-cluster-test:
> go test -v count=1 ./cmd/service-controller -use-cluster

will do

File: cmd/service-controller/definition_monitor_test.go (hunk 1 above)
Comment (nicob87) [reply cmd/service-controller/definition_monitor_test.go:196]: I am ok with that!

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-VERSION := $(shell git describe --tags --dirty=-modified)
+VERSION := $(shell git describe --tags --dirty=-modified --always)
```
Comment (fgiorgetti) [inline Makefile:1]: @nicob87 I have added this extra flag here, as I was constantly getting
errors in this particular line. Please see if you're ok with it.

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -42,3 +43,18 @@ func StringifySelector(labels map[string]string) string {
 	}
 	return result
 }
+
+// LabelToMap expects label string to be a comma separated
+// list of key and value pairs delimited by equals.
+func LabelToMap(label string) map[string]string {
```
Comment (fgiorgetti) [inline pkg/utils/utils.go:49]: @ajssmith this is the function that I added to pkg/utils.

**Pull request merged**